### PR TITLE
PLT-887 expose market data spec parser as utility

### PR DESCRIPTION
**Pull request by pondmouse**

### CODE REVIEW

File: projects/OG-Engine/src/main/java/com/opengamma/engine/marketdata/spec/MarketDataSpecificationParser.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ */
+package com.opengamma.engine.marketdata.spec;
+
+import org.threeten.bp.LocalDate;
+import org.threeten.bp.format.DateTimeParseException;
+
+import com.opengamma.id.UniqueId;
+
+/**
+ * Parses a string to produce a {@link MarketDataSpecification}.
```
Comment (cjkent) [inline projects/OG-Engine/src/main/java/com/opengamma/engine/marketdata/spec/MarketDataSpecificationParser.java]: Would be good to include a list of supported formats for the input string here.

File: projects/OG-Engine/src/main/java/com/opengamma/engine/marketdata/spec/MarketDataSpecificationParser.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+ */
+public class MarketDataSpecificationParser {
+
+  private static final String LIVE = "live";
+  private static final String SNAPSHOT = "snapshot";
+  private static final String FIXED_HISTORICAL = "fixedhistorical";
+  private static final String LATEST_HISTORICAL = "latesthistorical";
+
+  /**
+   * Parses a string to produce a {@link MarketDataSpecification}. See the output of {@link #getUsageMessage()}
```
Comment (cjkent) [inline projects/OG-Engine/src/main/java/com/opengamma/engine/marketdata/spec/MarketDataSpecificationParser.java]: Or maybe here.

File: projects/OG-Engine/src/main/java/com/opengamma/engine/marketdata/spec/MarketDataSpecificationParser.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+import org.threeten.bp.LocalDate;
+import org.threeten.bp.format.DateTimeParseException;
+
+import com.opengamma.id.UniqueId;
+
+/**
+ * Parses a string to produce a {@link MarketDataSpecification}.
+ */
+public class MarketDataSpecificationParser {
```
Comment (cjkent) [inline projects/OG-Engine/src/main/java/com/opengamma/engine/marketdata/spec/MarketDataSpecificationParser.java]: Should have a private constructor as there is no state and nothing but static methods.

File: projects/OG-Engine/src/main/java/com/opengamma/engine/marketdata/spec/MarketDataSpecificationParser.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      throw new IllegalArgumentException(specStr + " doesn't match 'live:source name'");
+    }
+    return LiveMarketDataSpecification.of(sourceName);
+  }
+
+  private static String removePrefix(String specStr, String prefix) {
+    return specStr.substring(prefix.length(), specStr.length());
+  }
+
+  public static String getUsageMessage() {
```
Comment (cjkent) [inline projects/OG-Engine/src/main/java/com/opengamma/engine/marketdata/spec/MarketDataSpecificationParser.java]: This should probably be private

**Comment by cjkent** [comment]
That's everything from me

File: projects/OG-Engine/src/main/java/com/opengamma/engine/marketdata/spec/MarketDataSpecificationParser.java (hunk 4 above)
Comment (pondmouse) [inline projects/OG-Engine/src/main/java/com/opengamma/engine/marketdata/spec/MarketDataSpecificationParser.java]: This is used in the SimulationTool when logging errors

**Pull request merged**